rxjava-support/src/main/java/tech/picnic/rx/RxSpring4Util.java

```diff
@@ -320,10 +320,14 @@ private RxSpring4Util() {}
       Duration heartbeat,
       LongFunction<T> heartbeatMapping,
       Scheduler scheduler) {
-    return flowable ->
-        Flowable.interval(heartbeat.toMillis(), MILLISECONDS, scheduler)
-            .map(heartbeatMapping::apply)
-            .mergeWith(flowable)
+    return publisher ->
+        Flowable.<T>fromPublisher(publisher)
```

Comment:
Nice, this issue was in the downcasting of `Flowable` -> `Publisher` in the function signature?

Reply:
Nope, the issue was that `mergeWith` doesn't complete until both sources complete. The `fromPublisher` here allows the source to be viewed as a `Flowable` in the two places that require this.

(In my testing the `#publish` call didn't seem necessary, but this way it is clearer that both the `mergeWith` and `takeUntil` operate on the same set of emissions.)

Reply:
Just thinking, won't this incur multiple subscriptions on `flowable` (the second in `takeUntil`) given that it's emissions aren't replayed?

Reply:
My bad, `publish` covers this :)

Reply:
My bad, this is covered by the `publish`